handlers/github_api_v1/handler/init.rb

```diff
@@ -0,0 +1,98 @@
+# Require the dependencies file to load the vendor libraries
+require File.expand_path(File.join(File.dirname(__FILE__), "dependencies"))
+
+class GithubApiV1
+  # ==== Parameters
+  # * +input+ - The String of Xml that was built by evaluating the node.xml handler template.
+  def initialize(input)
+    # Set the input document attribute
+    @input_document = REXML::Document.new(input)
+
+    # Retrieve all of the handler info values and store them in a hash variable named @info_values.
+    @info_values = {}
+    REXML::XPath.each(@input_document, "/handler/infos/info") do |item|
+      @info_values[item.attributes["name"]] = item.text.to_s.strip
+    end
+
+    # Retrieve all of the handler parameters and store them in a hash variable named @parameters.
+    @parameters = {}
+    REXML::XPath.each(@input_document, "/handler/parameters/parameter") do |item|
+      @parameters[item.attributes["name"]] = item.text.to_s.strip
+    end
+
+    @debug_logging_enabled = ["yes","true"].include?(@info_values['enable_debug_logging'].downcase)
+    @error_handling = @parameters["error_handling"]
+
+    @api_location = "https://api.github.com"
+    @access_token = @info_values["access_token"]
+
+    @body = @parameters["body"].to_s.empty? ? {} : JSON.parse(@parameters["body"])
+    @method = (@parameters["method"] || :get).downcase.to_sym
+    @path = @parameters["path"]
+    @path = "/#{@path}" if !@path.start_with?("/")
+
+    @accept = :json
+    @content_type = :json
+  end
+
+  def execute
+    # Initialize return data
+    error_message = nil
+    error_key = nil
+    response_code = nil
+    max_retries = 5
+    retries = 0
+
+    begin
+      api_route = "#{@api_location}#{@path}"
+      puts "API ROUTE: #{@method.to_s.upcase} #{api_route}" if @debug_logging_enabled
+      puts "BODY: \n #{@body}" if @debug_logging_enabled
+
+      response = RestClient::Request.execute \
+        method: @method, \
+        url: api_route, \
+        payload: @body.to_json, \
+        headers: {
+          :content_type => @content_type, 
+          :accept => @accept, 
+          :Authorization => "token #{@access_token}"
+        }
+      response_code = response.code
+    rescue RestClient::Exception => e
+      response_code = e.response.code
+
+      puts "There was an error parsing the JSON error response" if @debug_logging_enabled
```

Comment:
This error message is wrong. You are rescuing a RestClient exception, not a JSON Parse exception.

Reply:
Good catch, thanks

Reply:
Changed message to be accurate.